addons/delivery/wizard/choose_delivery_package.py

```diff
@@ -45,6 +45,15 @@ def _default_shipping_weight(self):
             total_weight = sum([po.qty_done * po.product_id.weight for po in move_line_ids])
             return total_weight
 
+    @api.onchange('delivery_packaging_id', 'shipping_weight')
+    def _onchange_packaging_weight(self):
+        if self.delivery_packaging_id.max_weight and self.shipping_weight > self.delivery_packaging_id.max_weight:
+            warning_mess = {
+                'title': _('Package too heavy!'),
+                'message': _('Your package weighs heavier than the max weight authorized')
```

Comment:
"The weight of your package is higher than the maximum weight authorized for this package type. Please choose another package type."